### ISSUE TITLE
Allow specifying column list for foreign key ON DELETE SET actions

### ISSUE THREAD

**Issue by omidh28** [prisma/prisma#14218]
## Problem

There are situations where a table references another table by more than one column (compound foreign key).  
For example: 

```prisma
model Department {
  id          String           @default(cuid())
  namespaceId String
  strategy    Strategy?  @relation(fields: [namespaceId, strategyId], references: [namespaceId, id], onDelete: SetNull)
  strategyId  String?

  @@id([namespaceId, id])
}

model Strategy {
  id          String           @default(cuid())
  namespaceId String

  @@id([namespaceId, id])
}

```

As you can see I want to set `Department.strategyId` to `null` when referenced record is deleted from `Strategy` table. However the database also tries to set the `department.namespaceId` to `null` but that column is a shared and NOT NULLABLE column. So the operation will fail.

## Suggested solution

As mentioned [here](https://git.postgresql.org/gitweb/?p=postgresql.git;a=commit;h=d6f96ed94e73052f99a2e545ed17a8b2fdc1fb8a), recent PostgreSQL versions support specifying which columns must be set to null when referenced record is deleted.

So it would be great if Prisma also supported it with something like this 
```prisma
onDelete: SetNull(strategyId)
```
However it's not always necessary to explicitly declare this because for example in my situation because `Department.namespaceId` is not nullable but `Department.strategyId` is, then Prisma can detect the problem automatically which can warn user when migrating or fix it implicitly by changing SQL file to something like this:

```
     CREATE TABLE "Department" (
        ...
        FOREIGN KEY (namespaceId, strategyId) REFERENCES Strategy ON DELETE SET NULL (strategyId)
    );
```

## Alternatives

Without suggested solution, I have to add extra column to my `Department` table: 
(Which still does not fully satisfy my requirements because I need to have some kind of constraints now to be sure both `namespaceId` and `strategyNamespaceId` are equal when defined.)

```prisma
model Department {
  id          String           @default(cuid())
  namespaceId String
  strategyNamespaceId String?
  strategy    Strategy?  @relation(fields: [strategyNamespaceId, strategyId], references: [namespaceId, id], onDelete: SetNull)
  strategyId  String?

  @@id([namespaceId, id])
}
```

**Comment by janpio** [comment]
Oh wow, interesting feature. Do you have a documentation link for PostgreSQL for this?
That this is quite a new feature will make supporting this reliably quite a challenge as Prisma itself is mostly version unaware (for now).

**Comment by omidh28** [comment]
Yes it is very new and just released in PostgreSQL 15 beta. This feature has been mentioned in [this page](https://www.postgresql.org/docs/15/sql-createtable.html):

> SET NULL [ ( column_name [, ... ] ) ]
> 
> Set all of the referencing columns, or a specified subset of the referencing columns, to null. A subset of columns can only be specified for ON DELETE actions.

It also works for `SET DEFAULT`: 

> SET DEFAULT [ ( column_name [, ... ] ) ]
> 
> Set all of the referencing columns, or a specified subset of the referencing columns, to their default values. A subset of columns can only be specified for ON DELETE actions. (There must be a row in the referenced table matching the default values, if they are not null, or the operation will fail.)

I know this feature probably takes a long time to be implemented in Prisma however It's very useful.
The weird thing is PostgreSQL during migration, does not complain about my tables schema that I've mentioned in my first post.

**Comment by omidh28** [comment]
Previously we could use this PostgreSQL feature by editing migration files manually, but now Prisma 4 complains about the schema itself:

> Error parsing attribute "@relation": The `onDelete` referential action of a relation must not be set to `SetNull` when a referenced field is required.
> Either choose another referential action, or make the referenced fields optional.
>

**Comment by Upsidedly** [comment]
any updates?
